Automattic/wp-calypso
# Sharing: Remove connections list library

This PR removes the connections-list module that #8991 made obsolete.

To test:
Navigate to My Sites > Sharing and verify connections can still be managed,
In the Editor, make sure the sharing accordion still works as expected.
